clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumer.java

```diff
@@ -197,6 +197,14 @@ public interface VeniceChangelogConsumer<K, V> {
    */
   Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> poll(long timeoutInMs);
 
+  /**
+   * Checks whether all subscribed partitions are caught up during bootstrap.
+   * @param offsetThresholdTimestamp Offset timestamp in millisecond. If a partition's (startTimestamp - heartbeatTimestamp)
+   *                                 is smaller or equal to offsetThresholdTimestamp, we consider this partition is caught up.
```

Comment:
I am very confused by this... is this an offset threshold or a timestamp threshold? It's better not to mix these terms in the same param name.

Also, I am confused about `startTimestamp - heartbeatTimestamp`... that would be a negative value, right?

Can we eliminate that param completely?

Reply:
Yeah I can update the naming. 
The idea is that the customer wants to be able to know if all the partitions are caught up by comparing HB TS with consumer startup TS. 
The passed in param in the API is a threshold, meaning  startTimestamp - heartbeatTimestamp < threshold we are good to announce bootstrap completed. 

With we continue to ingest, after bootstrap, the number in the map will the negative, but that should be fine as user will not need them anymore

Reply:
That is confusing, IMHO... let's say I pass zero into this param, but the consumer is quite behind for some reason. An hour after starting up, the consumer finally gets to the point where it is consuming the messages from an hour ago. At that point `isCaughtUp(0)` would finally return true. But is this useful? Why not eliminate the param altogether, and say that `isCaughtUp()` returns true if the most recent timestamp is <1m old, or something like that?

Reply:
I get what you mean, you think it is not necessary to have the param but use 1 min as constant.

There is only exactly one use case on this, our current observation is the bootstrap time should be around 1min. But certain partitions have not enough write volume.  The customer want to test with different parameter to see if certain param speed up the bootstrap. They potentially can tolerate messages like 5 mins old or something, so bootstrap can go even a bit faster. @ZacAttack you probably have context to this, maybe could you also chime in on this param thing?

Reply:
Also @zx0401 if you want to share anything from client side.

(Because I am ....... kind of convinced by what Felix mentioned above, a constant 1min threshold should be good for caught up logic)

Reply:
Checked with the partner, they are fine with constant threshold. That makes this easy. I am going to address the issue and update the PR.